### ISSUE TITLE
rll_planning_project build failed

### ISSUE THREAD

**Issue by oceanfly** [KITrobotics/rll_path_planning_project#2]
Hi,

I am trying to build this package but it showed below error as " "‘rll_msgs::PickPlace’ has not been declared". could you please suggest? thanks ! 

Errors     << rll_planning_project:make /home/tao/catkin_ws/logs/rll_planning_project/build.make.000.log
/home/tao/catkin_ws/src/rll_planning_project/src/planning_iface.cpp: In member function ‘bool PlanningIfaceBase::runPlannerOnce(const JobEnvGoalConstPtr&, rll_msgs::JobEnvResult*)’:
/home/tao/catkin_ws/src/rll_planning_project/src/planning_iface.cpp:188:13: error: ‘rll_msgs::PickPlace’ has not been declared
   rll_msgs::PickPlace::Request pick_place_req;

**Comment by wiewo** [comment]
Hi,

do you also have the [rll_sdk](https://github.com/KITrobotics/rll_sdk) in your catkin workspace? There are also detailed instructions in the RLL docs: https://rll-doc.ipr.kit.edu/manual_installation.html

And for setting up the planning project: https://rll-doc.ipr.kit.edu/planning_project.html

**Comment by oceanfly** [comment]
Hi Wiewo, thank you very much for your reply! Yes, I have the rll_sdk in my catkin workspace. All other modules were built successfully. Only this planning project module failed. The major complain is:

"rll_msgs::PickPlace has not been declared"

I am on ubuntu 18.04. Let me try again with your instructions and let you know. Thanks a lot!

**Comment by destogl** [comment]
Hi @oceanfly did you sources your workspace after compilation?

Can you show the message with ˋrosmsg showˋ?

**Comment by oceanfly** [comment]
Hi destogl and wiewo,
Thank you very much for your help! Here is what I have on my machine:

tao@tao-desktop:~/rll_ws$ ls src
rll_planning_project  rll_robot_playground_project  rll_sdk
tao@tao-desktop:~/rll_ws$ catkin build
-----------------------------------------------------
Profile:                     default
Extending:          [cached] /opt/ros/melodic
Workspace:                   /home/tao/rll_ws
-----------------------------------------------------
Build Space:        [exists] /home/tao/rll_ws/build
Devel Space:        [exists] /home/tao/rll_ws/devel
Install Space:      [unused] /home/tao/rll_ws/install
Log Space:          [exists] /home/tao/rll_ws/logs
Source Space:       [exists] /home/tao/rll_ws/src
DESTDIR:            [unused] None
-----------------------------------------------------
Devel Space Layout:          linked
Install Space Layout:        None
-----------------------------------------------------
Additional CMake Args:       None
Additional Make Args:        None
Additional catkin Make Args: None
Internal Make Job Server:    True
Cache Job Environments:      False
-----------------------------------------------------
Whitelisted Packages:        None
Blacklisted Packages:        None
-----------------------------------------------------
Workspace configuration appears valid.
-----------------------------------------------------
[build] Found '16' packages in 0.0 seconds.                                    
[build] Package table is up to date.                                           
Starting  >>> gazebo_version_helpers                                           
Starting  >>> rll_control                                                      
Starting  >>> rll_core                                                         
Starting  >>> rll_description                                                  
Finished  <<< rll_description                             [ 0.1 seconds ]      
Finished  <<< rll_control                                 [ 0.1 seconds ]      
Starting  >>> rll_gazebo                                                       
Finished  <<< rll_core                                    [ 0.1 seconds ]      
Starting  >>> rll_kinematics                                                   
Finished  <<< rll_gazebo                                  [ 0.1 seconds ]      
Finished  <<< rll_kinematics                              [ 0.2 seconds ]      
Starting  >>> rll_msgs                                                         
Starting  >>> rll_moveit_kinematics_plugin                                     
Finished  <<< gazebo_version_helpers                      [ 0.5 seconds ]      
Starting  >>> gazebo_grasp_plugin                                              
Finished  <<< rll_moveit_kinematics_plugin                [ 0.2 seconds ]      
Finished  <<< rll_msgs                                    [ 0.5 seconds ]      
Starting  >>> rll_moveit_config                                                
Starting  >>> rll_tools                                                        
Finished  <<< gazebo_grasp_plugin                         [ 0.5 seconds ]      
Finished  <<< rll_moveit_config                           [ 0.1 seconds ]      
Starting  >>> rll_move                                                         
Finished  <<< rll_tools                                   [ 0.1 seconds ]      
Finished  <<< rll_move                                    [ 0.6 seconds ]      
Starting  >>> rll_move_client                                                  
Finished  <<< rll_move_client                             [ 0.5 seconds ]      
Starting  >>> rll_planning_project                                             
Starting  >>> rll_robot_playground_project                                     
Finished  <<< rll_robot_playground_project                [ 0.2 seconds ]      
_______________________________________________________________________________
Errors     << rll_planning_project:make /home/tao/rll_ws/logs/rll_planning_project/build.make.001.log
/home/tao/rll_ws/src/rll_planning_project/src/planning_iface.cpp: In member function ‘bool PlanningIfaceBase::runPlannerOnce(const JobEnvGoalConstPtr&, rll_msgs::JobEnvResult*)’:
/home/tao/rll_ws/src/rll_planning_project/src/planning_iface.cpp:188:13: error: ‘rll_msgs::PickPlace’ has not been declared
   rll_msgs::PickPlace::Request pick_place_req;
             ^~~~~~~~~
/home/tao/rll_ws/src/rll_planning_project/src/planning_iface.cpp:189:13: error: ‘rll_msgs::PickPlace’ has not been declared
   rll_msgs::PickPlace::Response pick_place_resp;
             ^~~~~~~~~
/home/tao/rll_ws/src/rll_planning_project/src/planning_iface.cpp:213:3: error: ‘pick_place_req’ was not declared in this scope
   pick_place_req.pose_above = start_pose_above_;
   ^~~~~~~~~~~~~~
/home/tao/rll_ws/src/rll_planning_project/src/planning_iface.cpp:217:43: error: ‘pick_place_resp’ was not declared in this scope
   error_code = pickPlace(pick_place_req, &pick_place_resp);
                                           ^~~~~~~~~~~~~~~
/home/tao/rll_ws/src/rll_planning_project/src/planning_iface.cpp:217:16: error: ‘pickPlace’ was not declared in this scope
   error_code = pickPlace(pick_place_req, &pick_place_resp);
                ^~~~~~~~~
/home/tao/rll_ws/src/rll_planning_project/src/planning_iface.cpp: In member function ‘virtual RLLErrorCode PlanningIfaceBase::idle()’:
/home/tao/rll_ws/src/rll_planning_project/src/planning_iface.cpp:265:13: error: ‘rll_msgs::PickPlace’ has not been declared
   rll_msgs::PickPlace::Request pick_place_req;
             ^~~~~~~~~
/home/tao/rll_ws/src/rll_planning_project/src/planning_iface.cpp:266:13: error: ‘rll_msgs::PickPlace’ has not been declared
   rll_msgs::PickPlace::Response pick_place_resp;
             ^~~~~~~~~
/home/tao/rll_ws/src/rll_planning_project/src/planning_iface.cpp:281:5: error: ‘pick_place_req’ was not declared in this scope
     pick_place_req.pose_above = goal_pose_above_;
     ^~~~~~~~~~~~~~
/home/tao/rll_ws/src/rll_planning_project/src/planning_iface.cpp:285:45: error: ‘pick_place_resp’ was not declared in this scope
     error_code = pickPlace(pick_place_req, &pick_place_resp);
                                             ^~~~~~~~~~~~~~~
/home/tao/rll_ws/src/rll_planning_project/src/planning_iface.cpp:285:18: error: ‘pickPlace’ was not declared in this scope
     error_code = pickPlace(pick_place_req, &pick_place_resp);
                  ^~~~~~~~~
/home/tao/rll_ws/src/rll_planning_project/src/planning_iface.cpp: In member function ‘bool PlanningIfaceBase::getStartGoalSrv(rll_planning_project::GetStartGoal::Request&, rll_planning_project::GetStartGoal::Response&)’:
/home/tao/rll_ws/src/rll_planning_project/src/planning_iface.cpp:331:29: error: ‘beforeNonMovementServiceCall’ was not declared in this scope
   RLLErrorCode error_code = beforeNonMovementServiceCall(GET_START_GOAL_SRV_NAME);
                             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/tao/rll_ws/src/rll_planning_project/src/planning_iface.cpp:331:29: note: suggested alternative: ‘beforeServiceCall’
   RLLErrorCode error_code = beforeNonMovementServiceCall(GET_START_GOAL_SRV_NAME);
                             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
                             beforeServiceCall
/home/tao/rll_ws/src/rll_planning_project/src/planning_iface.cpp:339:16: error: ‘afterNonMovementServiceCall’ was not declared in this scope
   error_code = afterNonMovementServiceCall(GET_START_GOAL_SRV_NAME, error_code);
                ^~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/tao/rll_ws/src/rll_planning_project/src/planning_iface.cpp:339:16: note: suggested alternative: ‘afterServiceCall’
   error_code = afterNonMovementServiceCall(GET_START_GOAL_SRV_NAME, error_code);
                ^~~~~~~~~~~~~~~~~~~~~~~~~~~
                afterServiceCall
/home/tao/rll_ws/src/rll_planning_project/src/planning_iface.cpp: In member function ‘bool PlanningIfaceBase::checkPathSrv(rll_planning_project::CheckPath::Request&, rll_planning_project::CheckPath::Response&)’:
/home/tao/rll_ws/src/rll_planning_project/src/planning_iface.cpp:384:29: error: ‘beforeNonMovementServiceCall’ was not declared in this scope
   RLLErrorCode error_code = beforeNonMovementServiceCall(CHECK_PATH_SRV_NAME);
                             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/tao/rll_ws/src/rll_planning_project/src/planning_iface.cpp:384:29: note: suggested alternative: ‘beforeServiceCall’
   RLLErrorCode error_code = beforeNonMovementServiceCall(CHECK_PATH_SRV_NAME);
                             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
                             beforeServiceCall
/home/tao/rll_ws/src/rll_planning_project/src/planning_iface.cpp:391:16: error: ‘afterNonMovementServiceCall’ was not declared in this scope
   error_code = afterNonMovementServiceCall(CHECK_PATH_SRV_NAME, error_code);
                ^~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/tao/rll_ws/src/rll_planning_project/src/planning_iface.cpp:391:16: note: suggested alternative: ‘afterServiceCall’
   error_code = afterNonMovementServiceCall(CHECK_PATH_SRV_NAME, error_code);
                ^~~~~~~~~~~~~~~~~~~~~~~~~~~
                afterServiceCall
/home/tao/rll_ws/src/rll_planning_project/src/planning_iface.cpp: In member function ‘bool PlanningIfaceBase::checkGoalState()’:
/home/tao/rll_ws/src/rll_planning_project/src/planning_iface.cpp:474:13: error: ‘rll_msgs::PickPlace’ has not been declared
   rll_msgs::PickPlace::Request pick_place_req;
             ^~~~~~~~~
/home/tao/rll_ws/src/rll_planning_project/src/planning_iface.cpp:475:13: error: ‘rll_msgs::PickPlace’ has not been declared
   rll_msgs::PickPlace::Response pick_place_resp;
             ^~~~~~~~~
/home/tao/rll_ws/src/rll_planning_project/src/planning_iface.cpp:503:5: error: ‘pick_place_req’ was not declared in this scope
     pick_place_req.pose_above = goal_pose_above_;
     ^~~~~~~~~~~~~~
/home/tao/rll_ws/src/rll_planning_project/src/planning_iface.cpp:508:45: error: ‘pick_place_resp’ was not declared in this scope
     error_code = pickPlace(pick_place_req, &pick_place_resp);
                                             ^~~~~~~~~~~~~~~
/home/tao/rll_ws/src/rll_planning_project/src/planning_iface.cpp:508:18: error: ‘pickPlace’ was not declared in this scope
     error_code = pickPlace(pick_place_req, &pick_place_resp);
                  ^~~~~~~~~
/home/tao/rll_ws/src/rll_planning_project/src/planning_iface.cpp: In member function ‘RLLErrorCode PlanningIfaceBase::resetToStart()’:
/home/tao/rll_ws/src/rll_planning_project/src/planning_iface.cpp:581:13: error: ‘rll_msgs::PickPlace’ has not been declared
   rll_msgs::PickPlace::Request pick_place_req;
             ^~~~~~~~~
/home/tao/rll_ws/src/rll_planning_project/src/planning_iface.cpp:582:13: error: ‘rll_msgs::PickPlace’ has not been declared
   rll_msgs::PickPlace::Response pick_place_resp;
             ^~~~~~~~~
/home/tao/rll_ws/src/rll_planning_project/src/planning_iface.cpp:607:3: error: ‘pick_place_req’ was not declared in this scope
   pick_place_req.pose_above = start_pose_above_;
   ^~~~~~~~~~~~~~
/home/tao/rll_ws/src/rll_planning_project/src/planning_iface.cpp:612:43: error: ‘pick_place_resp’ was not declared in this scope
   error_code = pickPlace(pick_place_req, &pick_place_resp);
                                           ^~~~~~~~~~~~~~~
/home/tao/rll_ws/src/rll_planning_project/src/planning_iface.cpp:612:16: error: ‘pickPlace’ was not declared in this scope
   error_code = pickPlace(pick_place_req, &pick_place_resp);
                ^~~~~~~~~
make[2]: *** [CMakeFiles/planning_iface.dir/src/planning_iface.cpp.o] Error 1
make[1]: *** [CMakeFiles/planning_iface.dir/all] Error 2
make: *** [all] Error 2
cd /home/tao/rll_ws/build/rll_planning_project; catkin build --get-env rll_planning_project | catkin env -si  /usr/bin/make --jobserver-fds=6,7 -j; cd -
...............................................................................
Failed     << rll_planning_project:make                   [ Exited with code 2 ]
Failed    <<< rll_planning_project                        [ 6.7 seconds ]      
[build] Summary: 14 of 15 packages succeeded.                                  
[build]   Ignored:   1 packages were skipped or are blacklisted.               
[build]   Warnings:  None.                                                     
[build]   Abandoned: None.                                                     
[build]   Failed:    1 packages failed.                                        
[build] Runtime: 9.1 seconds total.                                            
tao@tao-desktop:~/rll_ws$ 


So the rll_planning_project package failed to be compiled. There is no info by running "rosmsg show".

Thanks!

**Comment by wiewo** [comment]
Please git pull the latest changes in the planning project repo. I pushed the latest changes on master to be compatible with the current SDK version.

**Comment by oceanfly** [comment]
Hi Wiewo, Thank you very much for your prompt reply! Yes, it works now! Thanks!
